### DISCUSSION TITLE
My IOS app stuck in publishing and getting build timeout after that

### DISCUSSION THREAD

**Discussion by abhidev54** [codemagic-ci-cd/codemagic-docs#2772] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1- Trigger IOS workflow build from codemagic
2- It pass many steps but stuck in publishing without any error
3- It says build timeout.

### Expected results

Even if it stuck, it should show proper error to investigate.

### Actual results

 Upload "/Users/builder/clone/build/ios/ipa/app.ipa" to App Store Connect

it stuck here and get build timeout without any error

### Build id (optional)

6702dc6c4531e53e3ee34651

**Comment by dtrdic** [comment, marked as answer]
hey @abhidev54 based on the logs, you have limited the maximum build duration of your builds to 6 minutes.
Please increase this number in your yaml file    ` max_build_duration: 6`
<img width="365" alt="image" src="https://github.com/user-attachments/assets/2323bb64-226d-4147-a712-f5c9f2897c21">

**Reply by abhidev54** [reply]
This is the issue, I increased the time to 10 minutes, it worked. Thanks for the help
